### DISCUSSION TITLE
Renovate removes GitLab approval rules independently of automerge option

### DISCUSSION THREAD

**Discussion by nc-igad** [renovatebot/renovate#31353] [Request Help] [answered]
### What would you like help with?

I think I found a bug

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab

### Please tell us more about your question or problem

Hi!
We recently upgraded our renovate version. After the upgrade, we saw that renovate started removing the approval rules on all merge requests independently of the value of `automerge`. 

The issue occurs starting from v37.421.7, I think this was introduced by this code change https://github.com/renovatebot/renovate/pull/29972.

**Example project configuration**
The renovate.json file of the project looks similar to this:
```
{
  "platformAutomerge": true,
  "gitLabIgnoreApprovals": true,
  "automergeType": "pr",
  "packageRules": [
    {
      "matchPackageNames": ["/azure/"],
      "automerge": true,
    }
  ]
}
```

**Expected behavior**
Renovate removes the approval rules on the new merge requests for azure packages only as automerge is enabled for them. All other merge requests are created with the projects default approval rules.

**Actual behavior**
Renovate creates the new merge requests and removes the approval rules on all of them.

**Comment by viceice** [comment, marked as answer]
definitely a bug, there's a missing condition 

- https://github.com/renovatebot/renovate/issues/31354
